Updated the getting start guide for Django (#6344)
getsentry/sentry-docs: src/platforms/python/guides/django/index.mdx

---
title: Django
redirect_from:
  - /clients/python/integrations/django/
  - /platforms/python/django/
description: "Learn about using Sentry with Django."
---

> The Django integration adds support for the [Django Framework](https://www.djangoproject.com/).
> 
> Don't already have an account and Sentry project established? Head over to [sentry.io](https://sentry.io), then return to this page.
> 
> <Note>
> 
> Sentry's Django integration enables automatic reporting of errors and exceptions as well as performance monitoring.
> 
> Django applications using Channels 2.0 will be correctly instrumented using Python 3.7. Older versions of Python require installation of [aiocontextvars](https://pypi.org/project/aiocontextvars/).
> 
> </Note>

Reviewer: ```suggestion
Sentry's [Django](https://www.djangoproject.com/) integration adds support for the Django Framework. It enables automatic reporting of errors and exceptions as well as performance monitoring. In order to get started using the integration, you should have a [Sentry account](https://sentry.io) and a project set up.

<Note>

If you're using Python 3.7., Django applications with Channels 2.0 will be correctly instrumented. Older versions of Python will require the installation of [aiocontextvars](https://pypi.org/project/aiocontextvars/).

</Note>
```

## Install

Install `sentry-sdk` from PyPI with the `django` extra:

```bash
pip install --upgrade 'sentry-sdk[django]'
```

## Configure

> To configure the Sentry SDK, initialize in your `settings.py` file.

Reviewer: ```suggestion
To configure the Sentry SDK, initialize it in your `settings.py` file:
```

```python {filename:settings.py}
import sentry_sdk

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",

    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    # We recommend adjusting this value in production,
    traces_sample_rate=1.0,
)
```

> ## Verify
> 
> This snippet includes an intentional error, so you can test that everything is working as soon as you set it up:
> 
> ```python
> from django.urls import path
> 
> def trigger_error(request):
>     division_by_zero = 1 / 0
> 
> urlpatterns = [
>     path('sentry-debug/', trigger_error),
>     # ...
> ]
> ```
> 
> Visiting this route will trigger an error that will be captured by Sentry.

Reviewer: ```suggestion
## Verify

The snippet below includes an intentional error that will be captured by Sentry when triggered. This will allow you to make sure that everything is working as soon as you set it up:

```python
from django.urls import path

def trigger_error(request):
    division_by_zero = 1 / 0

urlpatterns = [
    path('sentry-debug/', trigger_error),
    # ...
]
```

```

> ## Behaviour

Reviewer: ```suggestion
## Behavior
```
It would be good to add a line that introduces this section.

> ### Issue Reporting
> 
> - If you use `django.contrib.auth` and you have set `send_default_pii=True` in your call to `init`, user data (current user id, email address, username) is attached to error eventa.
> - Request data is attached to all events: **HTTP method, URL, headers, form data, JSON payloads**. Sentry excludes raw bodies and multipart file uploads.
> - Logs emitted by any logger are recorded as breadcrumbs by the [Logging](/platforms/python/guides/logging/) integration (this integration is enabled by default).
> 

Reviewer: ```suggestion
### Issue Reporting

- If you use `django.contrib.auth` and you've set `send_default_pii=True` in your call to `init`, user data (such as current user id, email address, username) will be attached to error events.
- Request data will be attached to all events: **HTTP method, URL, headers, form data, JSON payloads**. Sentry excludes raw bodies and multipart file uploads.
- Logs emitted by any logger will be recorded as breadcrumbs by the [Logging](/platforms/python/guides/logging/) integration (this integration is enabled by default).

```
